### ISSUE TITLE
Multiple chunks fail to render in 1.18.1

### ISSUE THREAD

**Issue by drnixon** [overviewer/Minecraft-Overviewer#2010]
Built overviewer from latest (7171af5) on macOS 12.1. Rendering into a clean directory after using --forceUpgrade on all chunks in a Spigot server. Getting a repeated error with multiple chunks:
```
ERROR:root:Could not render chunk -90,70 for some reason. This is likely a render primitive option error.                                                    ] 82678 106.43t/s eta 11h 24m 54s
ERROR:root:Full error was:
Traceback (most recent call last):
  File "/Users/[username]/Documents/MC/overviewer/Minecraft-Overviewer/overviewer_core/tileset.py", line 1091, in _render_rendertile
    c_overviewer.render_loop(
  File "/Users/[username]/Documents/MC/overviewer/Minecraft-Overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/Users/[username]/Documents/MC/overviewer/Minecraft-Overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/Users/[username]/Documents/MC/overviewer/Minecraft-Overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
```
Related to `chunky` now exceeding 255? End and nether seem to be rendering correctly, errors appear to be only in overworld renders so far.

**Comment by sirl1on** [comment]
I am getting the same error. No export is generated at all

- Vanilla Server
- World generated in 1.16, now at 1.18.1
- Using latest Docker Container setup without custom config

`docker run --rm -v ~/.minecraft/saves:/tmp/server/:ro -v $(pwd)/export/:/tmp/export/:rw -it marctv/minecraft-overviewer`

```
2021-12-22 09:51:15  Welcome to Minecraft Overviewer version 0.17.39 (2402e41)!
2021-12-22 09:51:16  Generating textures...
2021-12-22 09:51:25 E An error has occurred. This may be a bug. Please let us know!
See http://docs.overviewer.org/en/latest/index.html#help

This is the error that occurred:
Traceback (most recent call last):
  File "/tmp/overviewer/overviewer.py", line 653, in <module>
    ret = main()
  File "/tmp/overviewer/overviewer.py", line 562, in main
    tileSetOpts.update({"spawn": w.find_true_spawn()})  # TODO find a better way to do this
  File "/tmp/overviewer/overviewer_core/world.py", line 221, in find_true_spawn
    chunk = regionset.get_chunk(chunkX, chunkZ)
  File "/tmp/overviewer/overviewer_core/world.py", line 1637, in get_chunk
    level = data[1]['Level']
KeyError: 'Level'
```

**Comment by drnixon** [comment]
I did try running a render with greatmastermario's changes (#2012) but still getting the same errors during render. Never completes, map partially loads but never finishes rendering a lot of chunks. At this point in time players have visited many of the non-rendering areas, unclear if the errors are triggered by adjacent non-visited chunks or not.

**Comment by drnixon** [comment]
Addendum. Neglected to include. My generation script runs overviewer to update the map then runs it again to generate POI list. When trying to run overviewer to generate POI entries, this error:
```
2021-12-30 09:45:09 E An error has occurred. This may be a bug. Please let us know!
See http://docs.overviewer.org/en/latest/index.html#help

This is the error that occurred:
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/opt/homebrew/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/Users/[path-to-overviewer]/Minecraft-Overviewer/overviewer_core/aux_files/genPOI.py", line 132, in parseBucketChunks
    data = rset.get_chunk(b[0], b[1])
  File "/Users/[path-to-overviewer]/Minecraft-Overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/[path-to-overviewer]/Minecraft-Overviewer/overviewer.py", line 653, in <module>
    ret = main()
  File "/Users/[path-to-overviewer]/Minecraft-Overviewer/overviewer.py", line 168, in main
    g.genPOI.main()
  File "/Users/[path-to-overviewer]/Minecraft-Overviewer/overviewer_core/aux_files/genPOI.py", line 589, in main
    handleEntities(rset, config, args.config, rset_filters, markers)
  File "/Users/[path-to-overviewer]/Minecraft-Overviewer/overviewer_core/aux_files/genPOI.py", line 218, in handleEntities
    results = pool.map(parseBucketChunks, ((buck, rset, filters) for buck in buckets))
  File "/opt/homebrew/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/opt/homebrew/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 771, in get
    raise self._value
KeyError: 'Level'
```

**Comment by greatmastermario** [comment]
Copy-pasting what I put in another issue that mentions this same effect.

This is an issue caused by 1.18 updating the chunks to the new layout only after loading them in game. Once you get to certain distance from a 1.17 chunk, it will load as a proto-chunk, resetting its Status tag to "structure_starts". This results in Overviewer skipping this chunk since it thinks the chunk is not real. Normally this would be the case if the proto-chunk is on the edge of the world, but since this protochunk has existing data and is in the middle of the world, it results in the effects you see above. I do not know the proper way to handle this, but I can open a PR if we decide on a solution. See world.py line 1733-1735

**Comment by drnixon** [comment]
Aside from forcing a player to visit every single chunk I can't think of a way to address this without changing the way Overviewer handles these imported 1.17 chunks. My concern is that changing the way proto-chunks are handled just to handle edge cases for upgraded worlds seems like a lot of work for what may be a temporary problem.

**Comment by drnixon** [comment]
As a test I tried changing world.py to include proto-chunks in the list of chunks to render. It did generate a few chunks at edge of map (rendered in black, no lighting) but did not correct the issue with imported 1.17 chunks.

**Comment by drnixon** [comment]
Thinking about this more. Perhaps the correct way to handle it (without changing the way Overviewer normally runs) is to implement a conditional argument that will tell Overviewer to treat proto-chunks differently. This way if the program is called including an argument (such as `--update-pre118` or something) it would be able to handle these chunks but not change the way it runs otherwise.

**Comment by DarkArc** [comment]
> Thinking about this more. Perhaps the correct way to handle it (without changing the way Overviewer normally runs) is to implement a conditional argument that will tell Overviewer to treat proto-chunks differently. This way if the program is called including an argument (such as `--update-pre118` or something) it would be able to handle these chunks but not change the way it runs otherwise.

(Speaking as a user) Would this be in the config as well/configurable per map? I currently just invoke overviewer once to handle all my worlds so a CLI option wouldn't be ideal.

**Comment by drnixon** [comment]
My guess is one would only have to call this once for initial render. After that Overviewer would see the chunk as unchanged thus skip rendering it regardless. If the chunk updates after that, it would be because a player visited the chunk so the issue would no longer apply and Overviewer would update it as with any other changed chunk.

**Comment by drnixon** [comment]
Update. I re-ran Overviewer on the 1.18 world, into a clean directory. Repeated failure to render one chunk (-45, 60) with a few errors referencing adjacent or nearby chunks. After 10 failures Overviewer seems to freeze and no further progress is made, even when left to run overnight.

Investigating this shows that the chunk initially triggering this issue is empty. Not a proto-chunk, not an imported chunk, just empty. Opening world in MCASelector to view current world layout makes it clear - the area impacting my render is in the middle of the world, but is an area that has no content either imported from 1.17 or generated newly via visit in 1.18. Screenshot shows Overviewer output referencing the bad chunk (-45, 60) ... this corresponds to the center of the blank area showing in MCASelector in the same screenshot.
<img width="1345" alt="Screen Shot 2022-01-14 at 2 35 46 PM" src="https://user-images.githubusercontent.com/50930574/149582517-a7d882eb-a857-4fda-92d4-85430db2ee02.png">

**Comment by drnixon** [comment]
Visited chunk in question in game, then tried rendering again. Overviewer now choking on a different chunk, corresponding to another "hole" in the map. At least as far as I can tell, the error is being triggered in these open areas. Aside from manually visiting the chunks that trigger the error and trying again I am somewhat at a loss.

**Comment by drnixon** [comment]
Tried a different approach. Visited one of the offending chunks, placed and destroyed a block. Re-ran Overviewer. It breaks on the same chunk. This means the issue here is not necessarily due to the chunk being in a pre-loaded state. Attaching the .mca file containing the chunk (-90,70)
[r.-3.2.mca.zip](https://github.com/overviewer/Minecraft-Overviewer/files/7945287/r.-3.2.mca.zip)
.

**Comment by PremixedBones** [comment]
Just thought I'd chime in a say that I've been getting this same error. This world would render fine when it was 1.17.1, but now that it's been upgraded to 1.18.1 it triggers the same error message as others have posted in this thread:

> tileset.py:1105    12877 2022-01-31 09:05:05 ERROR    Could not render chunk -32,-816 for some reason. This is likely a render primitive option error.
tileset.py:1106    12877 2022-01-31 09:05:05 ERROR    Full error was:
Traceback (most recent call last):
  File "/home/itservices/src/Minecraft-Overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/itservices/src/Minecraft-Overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/itservices/src/Minecraft-Overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/itservices/src/Minecraft-Overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/itservices/src/Minecraft-Overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/itservices/src/Minecraft-Overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'

---------------------------------------
It's a Paper server, and I've verified there are no corrupt chunks, and even visited the one above and it loads just fine. Dug around in it (all the way down to former bedrock) and placed a bunch of blocks etc. It still refuses to render for some reason.

**Comment by Rysanlos** [comment]
Hello!
I'm getting the same repeated error with the last Windows binary 64bits.
```
ERROR:root:Could not render chunk 16,-16 for some reason. This is likely a render primitive option error.
ERROR:root:Could not render chunk 3,-4 for some reason. This is likely a render primitive option error.
ERROR:root:Full error was:
Traceback (most recent call last):
  File "overviewer_core\tileset.py", line 1093, in _render_rendertile
  File "overviewer_core\world.py", line 2120, in get_chunk
  File "overviewer_core\world.py", line 1935, in get_chunk
  File "overviewer_core\world.py", line 1713, in get_chunk
KeyError: 'Level'
```
The world was generated in 1.18.1 (it's new) and I'm using this configuration option:
```
worlds["Overworld"] = "D:/Documents/mcmap/worldbackup/world"
worlds["Nether"] = "D:/Documents/mcmap/worldbackup/world_nether"
worlds["End"] = "D:/Documents/mcmap/worldbackup/world_the_end"

renders["Overworld"] = {
    "world": "Overworld",
    "title": "Survival Day",
    "rendermode": smooth_lighting,
}

renders["Overworld Night"] = {
    "world": "Overworld",
    "title": "Survival Night",
    "rendermode": smooth_night,
}

renders["Nether"] = {
    "world": "Nether",
    "title": "Survival Nether",
    "rendermode": nether,
}

renders["End"] = {
    "world": "End",
    "title": "Survival End",
    "rendermode": normal,
}

outputdir = "D:/Documents/mcmap/render"
texturepath = "D:/Documents/mcmap/textures/textures.jar"
```
Is there anyway to ignore theses chunks errors and render anyway? Like a --ignore-error option?
Thank you

**Comment by drnixon** [comment]
I'm not sure what is going on with this issue but I cannot find a workaround. Resorted to deleting the chunks that were marked as unrenderable, but when I try to re-render - even with a clean target directory - Overviewer chokes at the same place and flags the same (now empty) chunks as a problem. Net result is that I don't have a usable map of any 1.18 world. I cannot tell if it is an error in the chunk itself, or some other issue with the script that triggers this error for specific random chunks.

**Comment by drnixon** [comment]
Seems like this is the same issue that cropped up in #1975 ??

**Comment by cliffmeyers** [comment]
I'm getting the same issue with a 1.18.1 Spigot world that was upgraded using `--forceUpgrade` a while back. 

Using latest `master` build

```
overviewer.py --version
Minecraft Overviewer 0.15.207 (7171af5)
built on Sat Apr 23 12:18:36 2022
```

Errors:

```
time overviewer.py --forcerender -c ./config.py
2022-04-23 12:34:24  Welcome to Minecraft Overviewer version 0.15.207 (7171af5)!
2022-04-23 12:34:24  Forcerender mode activated. ALL tiles will be rendered.
2022-04-23 12:34:24  Generating textures...
2022-04-23 12:34:38  Generating textures...
2022-04-23 12:34:52  Preprocessing...
2022-04-23 12:40:20  Rendering 956687 total tiles.
2022-04-23 12:45:50 E Could not render chunk 85,-59 for some reason. This is likely a render primitive option error.  ] 10018 30.44t/s eta 08h 38m 16s
2022-04-23 12:45:50 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
```

Similar to what I think a few others had seen, it's erroring on a chunk that's never been visited:

![world with unvisited chunks error](https://user-images.githubusercontent.com/143466/164923507-6dbbd746-1785-4c6e-9932-db9de363b7fb.png)

The world is quite large (~6 GB) so I haven't run it to completion yet to know if these errors actually break the map output. I've been using this map (with this same unvisited section) for a couple of years now and this is the first time I'd received this error from Overviewer.

Anything I can do to provide more useful info?

More detailed stack

```
time overviewer.py --forcerender -c ./config.py
2022-04-23 12:34:24  Welcome to Minecraft Overviewer version 0.15.207 (98a5f58)!
2022-04-23 12:34:24  Forcerender mode activated. ALL tiles will be rendered.
2022-04-23 12:34:24  Generating textures...
2022-04-23 12:34:38  Generating textures...
2022-04-23 12:34:52  Preprocessing...
2022-04-23 12:40:20  Rendering 956687 total tiles.
2022-04-23 12:45:50 E Could not render chunk 85,-59 for some reason. This is likely a render primitive option error.  ] 10018 30.44t/s eta 08h 38m 16s
2022-04-23 12:45:50 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
2022-04-23 12:45:50 E Could not render chunk 85,-59 for some reason. This is likely a render primitive option error.  ] 10044 30.47t/s eta 08h 37m 43s
2022-04-23 12:45:50 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
2022-04-23 12:45:52 E Could not render chunk 86,-58 for some reason. This is likely a render primitive option error.  ] 10097 30.54t/s eta 08h 36m 37s
2022-04-23 12:45:52 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
2022-04-23 12:46:00 E Could not render chunk 85,-59 for some reason. This is likely a render primitive option error.  ] 10475 30.87t/s eta 08h 30m 55s
2022-04-23 12:46:00 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
2022-04-23 12:46:01 E Could not render chunk 87,-59 for some reason. This is likely a render primitive option error.  ] 10503 30.89t/s eta 08h 30m 30s
2022-04-23 12:46:01 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
2022-04-23 12:46:01 E Could not render chunk 85,-59 for some reason. This is likely a render primitive option error.  ] 10530 30.91t/s eta 08h 30m 08s
2022-04-23 12:46:01 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
2022-04-23 12:46:02 E Could not render chunk 87,-59 for some reason. This is likely a render primitive option error.
2022-04-23 12:46:02 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
2022-04-23 12:46:03 E Could not render chunk 89,-59 for some reason. This is likely a render primitive option error.  ] 10557 30.91t/s eta 08h 30m 09s
2022-04-23 12:46:03 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
2022-04-23 12:46:04 E Could not render chunk 89,-59 for some reason. This is likely a render primitive option error.  ] 10585 30.91t/s eta 08h 30m 10s
2022-04-23 12:46:04 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
2022-04-23 12:46:05 E Could not render chunk 89,-59 for some reason. This is likely a render primitive option error.  ] 10612 30.87t/s eta 08h 30m 51s
2022-04-23 12:46:05 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
2022-04-23 12:46:07 E Could not render chunk 91,-59 for some reason. This is likely a render primitive option error.
2022-04-23 12:46:07 E Full error was:
Traceback (most recent call last):
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/tileset.py", line 1093, in _render_rendertile
    self.options['rendermode'], self.textures)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2049, in get_chunk
    return super(CroppedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 2120, in get_chunk
    retval = super(CachedRegionSet, self).get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1935, in get_chunk
    return self._r.get_chunk(x,z)
  File "/home/cmeyers/minecraft/oss/overviewer/overviewer_core/world.py", line 1713, in get_chunk
    chunk_data = chunk_data['Level']
KeyError: 'Level'
^C2022-04-23 12:46:16  Interrupted by user. Aborting.
```

**Comment by Zevac-ZSF** [comment]
Hi, as I have this same issue after upgrading my world to 1.18.2, I've been searching and testing far and long with my limited knowledge.
I did find that it turns out that when you run Minecraft with -forceupdate, it can actually NOT upgrade all chunks, it seems to be able to choke at one point without giving any message (usually on world with large, like >5 billon chunks), I've rerun the -forceupdate on my world and it did skip about 7 billion at first but then finished another 2 billion. Sadly that did not fix this problem. But it might be one of the reasons.

Second thing I found is that Mineways was also choking on some chunks, and after communication with the creator he told me that the inflate-function (using zlib) actually failed sometimes on its memory use, by default it seems to use 4 mb to inflate a chunk, but that is not always enough anymore. Maybe that is the reason here. I'm no programmer so I have no idea to research this, but I hope this info helps in solving this problem.

Z.

**Comment by Zevac-ZSF** [comment]
ow, and on an afterthought, would it be possible that when this error pops up, that chunk is simply left blank on the map (for now) but that it at least continues to render the rest? I have been unable to use my overviewer map for many months now and I really miss it :-) and I would already be very happy if it at least finishes its renders.

**Comment by Zevac-ZSF** [comment]
the creator of Mineways found out another issue with some of my chunks, it turns out some data has been written as integers instead of bytes. The cause might have been a bug in the lithum mod, which they said they fixed just before 1.18 came out. (https://www.curseforge.com/minecraft/mc-mods/lithium/files/3516367

Below is a screenshot of what he found.
![Chunk_withIncorrectValueTypes](https://user-images.githubusercontent.com/32144042/169161007-68ea2f7e-fd88-4a3f-a65a-c30ea2c0b758.png)
)

Is someone willing to research this? or am I just chasing ghosts? :-)

**Comment by Zevac-ZSF** [comment]
okay, I found the real cause, the field 'Level' inside a chunk in a .mca does not exist in certain situations. I've ran overviewers code with the addition of writing all chunks where this happens with, and it turns out all of them have the Status field set to 'empty', so they are skipped anyway.

Therefore one extra line fixes this issue, a check if the Level field actually exists:
<img width="641" alt="OverviewerFixForLevelErrorInChunks" src="https://user-images.githubusercontent.com/32144042/170713122-1399c523-c0a5-4d22-b8c2-bb300e910a38.PNG">

the extra line is on linenumber 1713
on lines 1732 and 1734 it is explained that if the Status field is filled with 'empty', those chunks will not be processed any further, so this fix has no further impact.

I am not a developer, so I have no real clue on how to use github.com to implement it myself so it can be validated/tested/implemented, is anybody interested to do that for me?

**Comment by drnixon** [comment]
Awesome. That does indeed fix the issue, world renders without problem after making the change and recompiling. Pull request made to get this edit into main branch.

**Comment by CounterPillow** [comment]
Please leave issues open until the PR that fixes them is merged.

**Comment by Zevac-ZSF** [comment]
Ok, thanx for implementing it drnixon. What is the next step to get a build (for me a windows build that is) to test/use this? I've ran some more testrenders and found no issues with this solution. But the renders via pycharm are quite slow and I would love to do a full run on my world again.

**Comment by drnixon** [comment]
You could pull my forked version but really it’s only the single change in world.py. But really if you’re going to recompile it I’d say start by pulling greatmastermario’s changes (#2012) as that also fixes world height issues in 1.18. Then make the edits to world.py, and build the resulting code per instructions on [overviewer.org](http://docs.overviewer.org/en/latest/building/). I’m on macOS so I’m always building from source. It looks more complicated than it really is, you just have to install the required support packages and then run the build script in the code.

**Comment by CounterPillow** [comment]
> I’d say start by pulling greatmastermario’s changes

already been merged

**Comment by Zevac-ZSF** [comment]
okay, sorry for sounding like a noob, but I never got the hang of how that goes with github.com, I basically downloaded the sources .zip from the overviewer site. And when I ran the build script I ended up with a overviewer....pyd, but couldn't find an overviewer.exe anywhere. so I started testing and running from within pycharm, which I had found by accident (I think I mentioned before, I am not a real developer, I am just quite good at reading code in any language lol).

what I am basically saying is that it is definetly not easy for everybody to 'build from source', even for a nerd like me, I feel like I am fumbling around in the dark, following information that seem to assume I know more then I do. I was just stubborn enough to keep trying and with that stumbled upon a solution.

So please, for the noobs like me and the non-nerds, just make a build we can download, unzip into a folder and run overviewer.exe :-)

**Comment by drnixon** [comment]
Sorry I can’t help there. As I said I’m on macOS so I’m calling the package via command line, not an .exe (and apparently I’m outdated as I didn’t realize the height changes were merged already - this render issue has stopped me from making any updates to my map since 1.18!)

The overviewer site has “windows beginner” instructions for running the program. Might help you. I can’t say whether building from source generates an .exe, I don’t have a windows platform to test it.

**Comment by cliffmeyers** [comment]
I just implemented this change, rebuilt and this is looking good! I am cranking out a full render of just over 1M tiles and giving my new 6900HX a proper stress test 😆 I'll report back once it's finished. Thanks @Zevac-ZSF @drnixon !!

(hopefully this isn't egregiously off topic)  @Zevac-ZSF if you're on Windows, I would recommend taking a look at Windows Subsystem for Linux (WSL). It is pretty straightforward to set up and makes it easy to install and launch a full-featured Ubuntu environment with a real shell and pretty good performance. I find that I can follow the default instructions for just about any project and it "just works." The docs and support for Windows tend to be pretty poor and it always felt like I was fighting uphill.

**Comment by CounterPillow** [comment]
I think I fixed this properly now, but I couldn't reproduce the crash with any sample files provided, so I don't know for sure.

Please note that our Windows builders are currently not working so it won't be in the Windows builds yet. In fact, basically none of our builders are working. Fun.
